### ISSUE TITLE
`nodefaults` channel should always be taken into account

### ISSUE THREAD

**Issue by maximlt** [conda/conda#12010]
### Checklist

- [X] I added a descriptive title
- [X] I searched open requests and couldn't find a duplicate

### What is the idea?

I'd like to be able to create environments whose channels are totally decoupled from the channels declared in the base environment. This is to be able to activate an environment and not having to care about the channels anymore to manage that environment, which is useful both locally and in CI systems. Note that the advantage of this approach is that this avoids users having to remember what channels an environment depend on (`conda install -c what?`) and to avoid the channels dance (installing a package leading to many upgrades of other packages due to the introduction of a new channel in the solver configuration).

I attempted to develop this approach the following way:

```
# This creates an empty environment
conda create -n myenv
conda activate myenv
# Configure the channels of this environment (not globally) using --env
conda config --env --append channels conda-forge
# Defaults is first in the list of channels, remove it
conda config --env --remove channels defaults
```

At this stage running `conda config --show-sources` shows:

```
==> /Users/mliquet/.condarc <==
auto_activate_base: False
channel_priority: disabled
channels:
  - defaults
always_yes: False
experimental_solver: libmamba

==> /Users/mliquet/miniconda3/envs/testconda/.condarc <==
channels:
  - conda-forge
```

So apparently the per-environment `.condarc` doesn't override the base `.condarc` environment. Indeed running `conda install fonts-anaconda` (this is only available in the *defaults* channel, not on *conda-forge*) does work, meaning *defaults* is part of the solve.

I could remove *defaults* from the base environment with `conda config --remove channels defaults` but that would touch at the global configuration which is certainly a bad idea. As far as I can see currently the only way to get this to work in an environment is to use `--override-channels`, e.g. `conda install fonts-anaconda -c conda-forge --override-channels` will terminate with a `PackagesNotFoundError`. This means that users have to repeat the channels they want to use and not forget to add `--override-channels`.

This behavior seems to be built in conda and reverting it would break for sure many environments and workflows. I suggest instead relying on another mechanism already available in conda to declare that the *defaults* channel should be ignored: `nodefaults`. One can add `nodefaults` in an `environment.yml` file to indicate conda not to use the *defaults* channel when creating the environment. In the example I showed above adding `nodefaults` to the environment configuration with `conda config --env --append channels nodefaults` has no effects whatsover, `conda install fonts-anaconda` still offers installing that package from *defaults*. I think for the sake of consistency, and for my specific use-case, `nodefaults` should always be taken into account.

**Comment by travishathaway** [comment]
Thanks for your feature request @maximlt. I personally think this is a great idea and would ensure better feature parity between the `conda` and `conda env` commands.

**Comment by jaimergp** [comment]
I think the undocumented `#!final` modifier might help here. See https://github.com/conda/conda/issues/11377

**Comment by maximlt** [comment]
Yep indeed it might help for my specific use case. However a consistent UX for `nodefaults` would I believe be a nice improvement too.

**Comment by maximlt** [comment]
Feature request still valid.
